app/routes/blog.$slug.tsx

```diff
@@ -9,16 +10,43 @@ import type { Mdx } from "blog/parser";
 export const loader = async function ({ params, request }: LoaderArgs) {
   const { slug } = params;
   invariant(typeof slug === "string", "Missing slug");
-  const post = (await fetch(
+  const post = await fetch(
     `${new URL(request.url).origin}/resource/get-blog-post/${slug}`
-  ).then((response) => response.json())) as Promise<Mdx>;
+  ).then((response) => response.json() as Promise<Mdx>);
   invariant(post !== undefined, "Could not find post");
-  return post;
+  return {
+    post,
+    url: request.url,
+  };
+};
+
+/**
+ * TODO: When we update to Remix v2, we need to change this function according to this
+ * page in the docs: https://remix.run/docs/en/1.16.0/pages/v2#route-meta
+ * Blog tags can't be added to meta in Remix v1. We need to wait until v2.
+ */
+export const meta: MetaFunction<typeof loader> = ({ data }) => {
+  const {
+    frontmatter: { title, author, date, imageUrl, description },
+  } = data.post;
+  const origin = new URL(data.url).origin;
+  return {
+    "og:url": data.url,
+    "og:type": "article",
+    "og:author": author,
+    "og:published_time": new Date(date).toISOString(),
+    "og:title": title,
+    "og:description": description,
+    "og:image": imageUrl[0] === "/" ? `${origin}${imageUrl}` : imageUrl,
+    "og:image:type": "image/jpeg",
+    "og:image:width": "1200",
+    "og:image:height": "630",
+  };
```

Comment:
Let's add the `title` and `description` tags for good measure 🙂 

```suggestion
  return {
    title,
    description,
    "og:url": data.url,
    "og:type": "article",
    "og:author": author,
    "og:published_time": new Date(date).toISOString(),
    "og:title": title,
    "og:description": description,
    "og:image": imageUrl[0] === "/" ? `${origin}${imageUrl}` : imageUrl,
    "og:image:type": "image/jpeg",
    "og:image:width": "1200",
    "og:image:height": "630",
  };
```